src/azul/deployment.py

```diff
@@ -427,8 +429,39 @@ def client(self, *args, **kwargs):
         Caching the result of this function is not necessary and will be harmful
         if the cached value is used by a thread other than the one that called
         this function.
+
+        Enable azul_logging in order to log Boto3 client's requests and
+        responses.
+        """
+        client = self.boto3_session.client(*args, **kwargs)
+        if azul_logging:
+            client.meta.events.register("before-send", self._log_client_request)
+            client.meta.events.register("response-received", self._log_client_response)
+        return client
+
+    def _log_client_request(self, event_name: str, request: Any, **kwargs) -> Optional[AWSResponse]:
+        """
+        Log the client's AWS API action, method, url being requested and the
+        body (if available). Logging of streamable body objects is skipped.
+        """
+        event_name = event_name.replace('before-send', 'Boto3 client')
```

Comment:
I think having a space in this string looks weird in the logs. Consider just "client" or "boto3".